### DISCUSSION TITLE
Velocity Rotation - self rotating when

### DISCUSSION THREAD

**Discussion by oivio** [Sixze/ALS-Refactored#295] [Q&A] [answered]
![rotat](https://user-images.githubusercontent.com/38628851/234964125-aad5caf2-ecf0-4b7d-ab48-ff5a3d130d2f.gif)

Gif is just example of the issue that I did come a cross.
Its only on velocity rotation. I am not sure if that is part of setting that I can change or it is a bug? So have to ask how to make player stop rotating when he move towards object or other actor?
It is really problematic when character is moving towards and trying lets say swing a sword but then player self rotating to the side.

**Comment by mshvern** [comment]
This is not really a bug per se, just a property of using velocity direction. 
The workaround is to change the calculation to use input direction instead.
Perhaps only when detecting a wall in front of the character. 

I don't think this repo needs to do anything about this, this is how ALS has always been (and how many games work). 
It is unfortunate that it ruins mantling in velocity direction though.

**Comment by oivio** [comment]
yea, I did look around Discord. Its common thing.  So solution would be as you mention to remove velocity direction or maybe create separate rotation mode just for control rotation or... I just found today that "roll" function has some bool for input direction. Maybe it would be possible to move that function.
Overall it would not be problematic but it is in my case. Character quickly self rotate when he collide other character.

**Comment by Sixze** [comment, marked as answer]
You can completely remove the impact of obstacles on velocity, by removing the `Velocity = (UpdatedComponent->GetComponentLocation() - OldLocation) / timeTick;` line from `AlsCharacterMovementComponent.cpp`

Also, the behavior of the character in your GIF is not actually the default behavior from ALS. You can return the default behavior by setting `bRotateTowardsDesiredVelocityInVelocityDirectionRotationMode` to false in the character settings, but the result will be even worse.

In the past, I tried to implement the "input direction" rotation mode, but the result was not satisfactory: the character's rotation was no longer synchronized with the velocity, and a quick change in the direction of the character’s movement looked terrible. That's why I added the `bRotateTowardsDesiredVelocityInVelocityDirectionRotationMode` setting as a compromise.

**Comment by oivio** [comment]
Thank you @Sixze for direction. That solution with disabling "Velocity = (UpdatedComponent->GetComponentLocation() - OldLocation) / timeTick;"  did help a lot in my case.
